warp10/src/main/java/io/warp10/script/functions/PIVOT.java

```diff
@@ -0,0 +1,302 @@
+      }
+    }
+    
+    top = stack.pop();
+    
+    if (!(top instanceof List) || 0 == ((List) top).size()) {
+      throw new WarpScriptException(getName() + " operates on a non empty list of labeling Geo Time Series™.");
+    }
+
+    List<Object> gts = (List<Object>) top;
+    
+    for (Object o: gts) {
+      if (!(o instanceof GeoTimeSerie)) {
+        throw new WarpScriptException(getName() + " operates on a list of labeling Geo Time Series™.");        
+      }
+      
+      //
+      // Check that none of the labeling GTS class names is already a label of one of the
+      // GTS to label
+      //
+      
+      GeoTimeSerie g = (GeoTimeSerie) o;
+
+      if (0 == GTSHelper.nvalues(g)) {
+        throw new WarpScriptException(getName() + " operates on non empty Geo Time Series™.");
+      }
+      
+      for (Entry<String,String> entry: g.getLabels().entrySet()) {
+        if (classes.contains(entry.getKey())) {
+          throw new WarpScriptException(getName() + " labeling class '" + entry.getKey() + "' is already a label of a Geo Time Series™ to label.");
+        }
+      }
+    }
+        
+    //
+    // Now sort all GTS
+    //
+    
+    for (Object g: gts) {
+      GTSHelper.sort((GeoTimeSerie) g);
+    }
+    
+    for (Object g: labeling) {
+      GTSHelper.sort((GeoTimeSerie) g);
+    }
+    
+    int[] labelingIndices = new int[labeling.size()];
+    int[] gtsIndices = new int[gts.size()];
+    
+    boolean done = false;
+    
+    Map<Map<String,String>, GeoTimeSerie> labeled = new HashMap<Map<String,String>, GeoTimeSerie>();
+    
+    //
+    long commonts = Long.MAX_VALUE;
+    
+    if (synchronous) {
+      commonts = GTSHelper.tickAtIndex((GeoTimeSerie) labeling.get(0), 0);
+    } else {
+      for (int i = 0; i < labelingIndices.length; i++) {
+        if (GTSHelper.tickAtIndex((GeoTimeSerie) labeling.get(i), 0) < commonts) {
+          commonts = GTSHelper.tickAtIndex((GeoTimeSerie) labeling.get(i), 0);
+        }
+      }
+    }
+
+    while (!done) {
+      boolean skip = false;
+      
+      if (synchronous) {        
+        // Advance labeling GTS indices so that all timestamps are equal
+        for (int i = 0; i < labelingIndices.length; i++) {
+          GeoTimeSerie g = (GeoTimeSerie) labeling.get(i);
+          int n = GTSHelper.nvalues(g);
+          while(labelingIndices[i] < n && GTSHelper.tickAtIndex(g, labelingIndices[i]) < commonts) {
+            labelingIndices[i]++;
+          }
+          if (labelingIndices[i] >= n) {
+            done = true;
+            skip = true;
+            break;
+          }
+          if (GTSHelper.tickAtIndex(g, labelingIndices[i]) > commonts) {
+            commonts = GTSHelper.tickAtIndex(g, labelingIndices[i]);
+            skip = true;
+            break;
+          }
+        }
+        
+        if (skip) {
+          continue;
+        }    
+        
+        // Increase the index of the first labeling GTS
+        labelingIndices[0]++;
```

Comment:
Too early, this index is used line 176 & 178.
Increment should be done line 261.